src/Features/Core/Portable/PdbSourceDocument/ImplementationAssemblyLookupService.cs

```diff
@@ -0,0 +1,196 @@
+
+            implementationDllPath = null;
+            return false;
+        }
+
+        public string? FollowTypeForwards(ISymbol symbol, string dllPath, IPdbSourceDocumentLogger? logger)
+        {
+            // If we find any type forwards we'll assume they're in the same directory
+            var basePath = Path.GetDirectoryName(dllPath);
+            if (basePath is null)
+                return dllPath;
+
+            // Only the top most containing type in the ExportedType table actually points to an assembly
+            // so no point looking for nested types.
+            var typeSymbol = MetadataAsSourceHelpers.GetTopLevelContainingNamedType(symbol);
+
+            try
+            {
+                var typeFullMetadataName = $"{typeSymbol.ContainingNamespace.MetadataName}.{typeSymbol.MetadataName}";
+                while (File.Exists(dllPath))
+                {
+                    // We try to use the cache to avoid loading the file
+                    if (_typeForwardCache.TryGetValue(dllPath, out var typeForwards))
+                    {
+                        if (typeForwards is not null && typeForwards.TryGetValue(typeFullMetadataName, out var cachedAssemblyName))
+                        {
+                            dllPath = Path.Combine(basePath, $"{cachedAssemblyName}.dll");
+                            logger?.Log(FeaturesResources.Symbol_found_in_assembly_path_0, dllPath);
+                            continue;
+                        }
+
+                        return dllPath;
+                    }
+
+                    using var fileStream = File.OpenRead(dllPath);
+                    using var reader = new PEReader(fileStream);
+                    var md = reader.GetMetadataReader();
+
+                    var assemblyName = GetExportedTypeAssemblyName(md, typeFullMetadataName, dllPath);
+                    if (assemblyName is null)
+                    {
+                        // Didn't find a type forward, so the current DLL is the right one to use
+                        return dllPath;
+                    }
+
+                    dllPath = Path.Combine(basePath, $"{assemblyName}.dll");
+                    logger?.Log(FeaturesResources.Symbol_found_in_assembly_path_0, dllPath);
+                }
+            }
+            catch (Exception ex) when (IOUtilities.IsNormalIOException(ex))
+            {
+            }
+
+            return null;
+        }
+
+        public void Clear()
+        {
+            _typeForwardCache.Clear();
+        }
+
+        private static bool TryNugetLibToRef(string referenceDllPath, [NotNullWhen(true)] out string? implementationDllPath)
+        {
+            implementationDllPath = null;
+
+            // For some nuget packages if the reference path has a "ref" folder in it, then the implementation assembly
+            // will be in the corresponding "lib" folder.
+            var start = referenceDllPath.IndexOf(@"\ref\");
```

Comment:
>\ref\ [](http://example.com/codeflow?start=51&length=5)

This won't work on Unix. Perhaps better to split the path as below? `referencedDllPath.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)`?

Reply:
Good point. This code was copied from a previous PR, but definitely should be updated